apps/meteor/server/models/raw/Roles.ts

```diff
@@ -164,6 +164,14 @@ export class RolesRaw extends BaseRaw<IRole> implements IRolesModel {
 		return this.find(query, options || {}) as P extends Pick<IRole, '_id'> ? Cursor<P> : Cursor<IRole>;
 	}
 
+	findByScope<P>(scope: IRole['scope'], options?: FindOneOptions<IRole>): P extends Pick<IRole, '_id'> ? Cursor<P> : Cursor<IRole> {
+		const query: FilterQuery<IRole> = {
+			scope,
+		};
+
+		return this.find(query, options || {}) as P extends Pick<IRole, '_id'> ? Cursor<P> : Cursor<IRole>;
+	}
```

Comment:
I'm not sure why other methods were created like this, but I'd say you don't need this generic `P` and all this crazy, you could simply do:
```suggestion
	findByScope(scope: IRole['scope'], options?: FindOneOptions<IRole>): Cursor<IRole> {
		const query = {
			scope,
		};

		return this.find(query, options || {});
	}
```